### PR TITLE
Add shared place for React components

### PR DESCRIPTION
**Pull request by y-kkamil**
**Description**

Changes proposed in this pull request:

- Introduce 'shared place' for react components

**Related issue(s)**
#1297

### CODE REVIEW

**Comment by y-kkamil** [comment]
The name 'szerdstuff' is of course, still to be agreed upon ;)

File: szerdstuff/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "jest",
+    "build": "webpack",
+    "start:kyma": "webpack --watch --config webpack-dev.config.js"
+  },
+  "author": "hasselhoff-team@kyma",
+  "dependencies": {
+    "fundamental-react": "^0.6.1",
+    "jest": "^24.9.0",
```
Comment (sjanota) [inline szerdstuff/package.json]: Isn't jest a dev dependency?

File: szerdstuff/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "react": "^16.10.2"
```
Comment (sjanota) [inline szerdstuff/package.json]: I'd make react a peer dependency

File: szerdstuff/package.json (hunk 1 above)
Comment (y-kkamil) [reply szerdstuff/package.json]: it is, i've change it

File: szerdstuff/package.json (hunk 2 above)
Comment (y-kkamil) [reply szerdstuff/package.json]: done

**Comment by y-kkamil** [comment]
> Works well 👍
> 
> Apart from 3 in-code comments, I've got one general question:
> `szerdstuff/services/.gitkeep` - what is it for?

I wanted to create a basic folder structure for our shared components/contexts/services, and since we don't have any service yet, and git does not track empty folders I had to put there an arbitrary file - it's just a convention, the file doesn't have any meaning. ;)

File: shared/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import 'core-js/stable';
+import 'regenerator-runtime/runtime';
```
Comment (parostatkiem) [inline shared/index.js]: Do we use any generator functions?

File: shared/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  "author": "hasselhoff-team@kyma",
+  "devDependencies": {
+    "@babel/core": "^7.6.4",
+    "@babel/plugin-proposal-class-properties": "^7.5.5",
+    "@babel/preset-env": "^7.6.3",
+    "babel-jest": "^24.9.0",
+    "core-js": "^3.3.2",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^24.9.0",
```
Comment (parostatkiem) [inline shared/package.json]: It has been defined on the upper level.

File: shared/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "node-sass": "^4.12.0",
```
Comment (parostatkiem) [inline shared/package.json]: It has been defined on the upper level.

File: shared/package.json [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    "sass-loader": "^8.0.0",
+    "webpack": "^4.41.2",
+    "webpack-bundle-analyzer": "^3.6.0",
+    "webpack-cli": "^3.3.9"
```
Comment (parostatkiem) [inline shared/package.json]: It has been defined on the upper level.

File: shared/index.js (hunk 3 above)
Comment (y-kkamil) [reply shared/index.js]: Apparently not, I have also removed core-js import, as it looks like it is not needed.

File: compass/Makefile [hunk 7]
```diff
@@ -2,14 +2,21 @@ IMG_NAME = compass-ui
...
+	make -C "../" ci-pr
+
+.PHONY: root-master
+root-master:
+	make -C "../" ci-master
 
 .PHONY: resolve
 resolve:
-	cd .. && npm ci
+	cd .. && npm ci && cd shared && npm ci && npm run build
```
[the remark this answers is not in the document]
Comment (y-kkamil) [reply compass/Makefile]: done.

File: shared/package.json (hunk 6 above)
Comment (y-kkamil) [reply shared/package.json]: removed

File: shared/package.json (hunk 5 above)
Comment (y-kkamil) [reply shared/package.json]: removed

**Pull request merged**